src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -906,6 +905,20 @@ private static Autogenerated.StateOptions.Types.StateConcurrency GetStateConcurr
             };
         }
 
+        private void AddDaprApiTokenHeader(Metadata headers)
+        {
+            if (!string.IsNullOrWhiteSpace(this.daprApiToken))
+            {
+                headers.Add("dapr-api-token", this.daprApiToken);
+                return;
+            }
+
+            var apiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+            if(!string.IsNullOrWhiteSpace(apiToken))
+            {
+                headers.Add("dapr-api-token", apiToken);
+            }
```

Comment:
Can we put this logic elsewhere? This logic appears in a bunch of places. Ideally the builder would be responsible for this, and the client would just do as its told.